### PR TITLE
feat: add container image scanning to ci and release

### PR DESCRIPTION
**Pull request by madpah**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,6 +43,54 @@ jobs:
...
+                  cache: true
+
+            # See https://help.sonatype.com/en/npm-application-analysis.html
+            - name: Install Production Node Dependencies
+              run: corepack yarn install --prod
+
+            - name: Login to Docker Hub
+              uses: docker/login-action@v3
+              with:
+                  username: ${{ secrets.DOCKER_HUB_USERNAME }}
```
Comment (bhamail) [inline .github/workflows/ci.yml:72]: I still ran into local build issues with my own user/token for docker hub here, but guessing that's a "me" problem

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -32,8 +36,31 @@ jobs:
                   username: ${{ secrets.DOCKER_HUB_USERNAME }}
                   password: ${{ secrets.DOCKER_HUB_TOKEN }}
 
-            # - name: Set Yarn Version
-            #   run: yarn version --new-version ${{ env.THIS_VERSION }}
+            - name: Build Container and export to Docker
```
Comment (bhamail) [inline .github/workflows/release.yml:39]: Not sure I understand if the current build ensures the go app (ci.yaml -> build) and react app (release.yml -> release) are both successfully built with the latest code. Meaning, we might need a "needs" directive somewhere in the CI configs to make sure a successful go build precedes the react/docker builds.

**Review by bhamail — APPROVED** [review]
A couple questions/comments, but looks good 'nuf to me.

File: .github/workflows/release.yml (hunk 2 above)
Comment (madpah) [reply .github/workflows/release.yml:39]: If go build fails (non-zero exit code), job would exit with failure if that's what you meant @bhamail?

Comment (bhamail) [reply .github/workflows/release.yml:39]: In GHActions, each "job" is "independent" of other jobs. So while the go build will fail with non-zero exit code, that failure would not prevent a separate run of the "release" job from succeeding. It seems like there is a possibility of deploying with a "stale" go app, but that possibility is unlikely.

I'm alluding to possible need to create some "dependency" between jobs where the "react app" job should possibly  "depend" on a successful build of the "go app", thereby ensuring the deployed docker image is always "up to date".
It's an edge case that may not be worth worrying about, but feels like it does exist in the current CI config.

**Pull request merged**